.github/workflows/build-artifact-wrapper.yaml

```diff
@@ -0,0 +1,71 @@
+name: "Build tt-metal artifacts"
+
+on:
+  workflow_call:
```

Comment:
Does this need `workflow_call`? This is just so that we can test `build-artifact` on branch via `workflow_dispatch` right? The way to call `build-artifact` in another workflow is to instead call `build-docker-artifact` + `build-artifact` , where `build-artifact` depends on `build-docker-artifact`?

Reply:
I was thinking it might be a good idea to replace mentions of `build-artifact.yaml` with `build-artifact-wrapper.yaml` since `build-docker-artifact` must usually always precede `build-artifact` (unless run on main branch)

Reply:
so if `build-artifact` shouldn't be called anymore, shouldn't `build-docker-artifact` be merged into `build-artifact` now? This reduces the amount of spam